### ISSUE TITLE
(API): S2321: Excessive stack depth comparing types 'any' and 'FindConditions<T>'.

### ISSUE THREAD

**Issue by xmlking** [xmlking/ngx-starter-kit#23]
with typescript 3.2.2, we see following ignorable error when API server is started `ng serve api`

typescript@3.1.6 works fine, typescript@3.2.2 show error

```
ERROR in undefined(undefined,undefined):
TS2321: Excessive stack depth comparing types 'any' and 'FindConditions<T>'.
Version: typescript 3.2.2
```

Depends on: 
https://github.com/Microsoft/TypeScript/issues/21592

**Comment by johannesschobel** [comment]
Dear @xmlking ,

i ran into the same error when i tried starting the API with `ng serve api`.. However, in my case (Windows), this error is **not ignorable**, i.e., i cannot access the api via `localhost:3000` nor can i access the docs via `localhost:3000/docs`

How can i get the API up and running?

All the best

**Comment by xmlking** [comment]
Glad you are testing on windows and helps identifying issues. 
To experience this monorepo development model, you can try this mini version of this repo 
https://github.com/xmlking/connect4
Hope you have tools  to start Postgres before starting API server. In case if you don’t have docker env, you can start Postgres with standard intalation

**Comment by johannesschobel** [comment]
Hey there, thanks a lot for the fast response on this issue..
Strangly, the `Connect4` demo works. Still, the mentioned error arises when i run `(npm run) ng serve api` .. Nest boots the entire application and then the error occurs.. However, the application is deployed and i can access the api and docs..

Do you know why?

**Comment by xmlking** [comment]
That is also what I see on Mac for this project...
May be you can try deleting node_modules and install again.
If you really want to make it work, switch to typescript 3.1.x and matching angular version, and try this repo...

**Comment by johannesschobel** [comment]
I am very sorry - i don't get the `switch to typescript 3.1.x` part.. Actually, neither repository (`connect4` nor `ngx-starter-kit`) uses `typescript 3.1.x`.. But the one runs, the other doesnt :confused:

**Comment by xmlking** [comment]
try this 
```bash
npm i typescript@3.1.6 -D 
```

**Comment by johannesschobel** [comment]
Ok - Update on this..
Actually, the problem is, that the mentioned `.flatMap()` function is only available on `NodeJS 11+`! I was running the latest LTS (`Node 10.15`) on my dev machine..

When upgrading to `Node 11.7` the error disappears and everything works :heart: Awesome!
Maybe we should add this information / requirement somewhere in the docs?! ;)
Really appreaciate your help on this!

**Comment by xmlking** [comment]
Great. Please checkout `PLAYBOOK.md`  file for all prerequisites and tested global npm modules.

**Comment by xmlking** [comment]
> Ok, i found out, that the following function
> 
> [ngx-starter-kit/apps/api/src/app/core/core.module.ts](https://github.com/xmlking/ngx-starter-kit/blob/5e101382ce747bde7d27300a200bdde92f8b1084/apps/api/src/app/core/core.module.ts#L13-L19)
> Lines 13 to 19 in [5e10138](/xmlking/ngx-starter-kit/commit/5e101382ce747bde7d27300a200bdde92f8b1084)
>  function requireAllClasses(rc) { 
>    return rc 
>      .keys() 
>      .filter(filePath => !filePath.includes('base')) 
>      .flatMap(key => Object.values(rc(key))) 
>      .filter(isClass); 
>  } 
> 
> is available in the `ngx-starter-kit` but **not** in the `connect4` package!
> What exactly is this function used for?

This solved a hard problem making typeorm/nestjs integration into nx workspace and provide seamless developer experience. Thanks to @tomastrajan 

https://twitter.com/tomastrajan/status/1079724753253003264?s=21
https://github.com/nrwl/nx/issues/803

**Comment by johannesschobel** [comment]
thanks a lot for the explanation!

**Comment by xmlking** [comment]
This will be fixed with typescript@3.7.0
